src/Avalonia.Controls/Platform/ExportAvaloniaModuleAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Avalonia.Platform
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public class ExportAvaloniaModuleAttribute : Attribute
+    {
+        public ExportAvaloniaModuleAttribute(string name, Type moduleType)
+        {
+            Name = name;
+            ModuleType = moduleType;
+        }
+
+        public string Name { get; private set; }
+        public Type ModuleType { get; private set; }
+
+        public string RequiredWindowingSubsystem { get; set; } = "";
```

Comment:
@kekekeks apologies, GitHub only showed me these comments after I'd finished my review. I *think* that you can do that with this system, is that correct @jkoritzinsky ?